src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1864,6 +1864,42 @@
   <data name="UsesPreviewTypeParameterMessage" xml:space="preserve">
     <value>'{0}' uses the preview type '{1}' and needs to opt into preview features. See {2} for more information.</value>
   </data>
+  <data name="ConstantExpectedApplicationTitle" xml:space="preserve">
+    <value>Incorrect use of ConstantExpectedAttribute</value>
+  </data>
+  <data name="ConstantExpectedApplicationDescription" xml:space="preserve">
+    <value>ConstantExpectedAttribute is not applied correctly on the paramter.</value>
+  </data>
+  <data name="ConstantExpectedUsageTitle" xml:space="preserve">
+    <value>Constant is expected for the parameter</value>
+  </data>
+  <data name="ConstantExpectedUsageDescription" xml:space="preserve">
+    <value>The parameter expects a constant for optimal performance.</value>
+  </data>
+  <data name="ConstantExpectedNotSupportedMessage" xml:space="preserve">
+    <value>The '{0}' parameter is not supported for ConstantExpectedAttribute.</value>
+  </data>
+  <data name="ConstantExpectedIncompatibleConstantTypeMessage" xml:space="preserve">
+    <value>The '{0}' value is not compatible with parameter type of '{1}'.</value>
+  </data>
+  <data name="ConstantExpectedInvalidBoundsMessage" xml:space="preserve">
+    <value>The '{0}' value does not fit within the parameter value bounds of '{1}' to '{2}'.</value>
```

Comment:
NIT: doesn't seem need to complicate by mentioning the bounds of a type
```suggestion
    <value>The '{0}' value is invalid for the parameter of '{1}' type </value>
```

Reply:
hmm but we can specify custom bounds

Reply:
hm, OK